src/widget/rating/rating.js

```diff
@@ -0,0 +1,117 @@
+import RangeWidget from '../../widget/range/range-widget';
+import { isNumber } from '../../js/utils';
+
+/**
+ * @extends RangeWidget
+ */
+class RatingWidget extends RangeWidget {
+    /**
+     * @type string
+     */
+    static get selector() {
+        return '.or-appearance-rating input[type="number"]';
+    }
+
+    _init() {
+        const fragment = document.createRange().createContextualFragment( this._getHtmlStr() );
+        this.element.after( fragment );
+        this.element.classList.add( 'hide' );
+        this.widget = this.question.querySelector( '.widget' );
+
+        let ticks = this.props.ticks ? Math.ceil( Math.abs( ( this.props.max - this.props.min ) / this.props.step ) ) : 1;
+        // Now reduce to a number < 50 to avoid showing a sold black tick line.
+        let divisor = Math.ceil( ticks / this.props.maxTicks );
+        while ( ticks % divisor && divisor < ticks ) {
+            divisor++;
+        }
+        ticks = ticks / divisor;
+
+        let addTicks = new Array( ticks ).fill( '<span></span>' ).join( '' );
+        let ratingStars = '',
+            name = Math.random().toString( 36 ).substring( 2, 15 );
+        for ( let i = this.props.min; i <= this.props.max; i += this.props.step ) {
+            ratingStars += `<input type=radio name="rating-stars__${name}" class="star rating__tick" value="${i}"/>`;
+        }
+        addTicks = ratingStars;
+
+        this.widget.querySelector( '.range-widget__ticks' )
+            .append( document.createRange().createContextualFragment( addTicks ) );
+
+        this.widgetThick = this.widget.querySelector( '.range-widget__ticks' );
+        this.widgetThick.querySelectorAll( '.star.rating__tick' ).forEach( el => {
+            el.addEventListener( 'click', () => {
+                const selected = this.widgetThick.querySelector( 'input:checked' );
+                // contains class 'empty' means first load
+                if ( ( this.value != selected.value || this.widgetThick.classList.contains( 'empty' ) ) ) {
+                    this.value = selected.value;
+                } else {
+                    this.value = '';
+                    selected.checked = false;
+                }
+            } );
+        } );
+
+        // loads the default value if exists
+        this.update();
+
+        if ( this.props.readonly ) {
+            this.disable();
+        }
+    }
+
+    /**
+     * This is separated so it can be extended (in the analog-scale widget)
+     *
+     * @return {string} HTML string
+     */
+    _getHtmlStr() {
+        return `<div class="widget range-widget">
```

Comment:
would `class="widget rating-widget"` work?

Reply:
Would be nice to eliminate all references in css and javascript (variables) to "ticks" and "range-*"

Reply:
code updated